sockeye/inference.py

```diff
@@ -1009,29 +1013,25 @@ def _beam_search(self,
                                                                        models_output_layer_w,
                                                                        models_output_layer_b)
 
-            # (2) compute length-normalized accumulated scores in place
-            if t == 1 and self.batch_size == 1:  # only one hypothesis at t==1
-                scores = scores[:1] / self.length_penalty(lengths[:1])
-            else:
-                # renormalize scores by length ...
-                scores = (scores + scores_accumulated * self.length_penalty(lengths - 1)) / self.length_penalty(lengths)
-                # ... but not for finished hyps.
-                # their predicted distribution is set to their accumulated scores at C.PAD_ID.
+            # (2) Replace finished rows with inf in all columns except column 0, which has the
+            # accumulated model score
+            if t > 1:
+                scores = scores + scores_accumulated
                 pad_dist[:, C.PAD_ID] = scores_accumulated
-                # this is equivalent to doing this in numpy:
```

Comment:
why remove this comment? I find it useful :)

Reply:
No prob.